src/QualificationRunner.Core/Services/QualificationRunner.cs

```diff
@@ -430,14 +430,22 @@ private IReadOnlyList<Input> retrieveInputs(dynamic qualificationPlan)
 
       private int? getSectionLevel(IReadOnlyList<dynamic> sections, int? sectionId, string sectionReference, int currentLevel = 1)
       {
+         //no sections. nothing to see here
          if (sections == null)
             return null;
 
-         var section = sections.FirstOrDefault(x => x.Id == sectionId || x.Reference == sectionReference);

```

Comment:
This was always returning the first level because one of this condition was correct.